smileidentity/flutter
# Add networking APIs

Story: https://app.shortcut.com/smileid/story/11109

## Summary

Creates networking functions via a new top level `SmileID.api` object, mirroring the interface in our native SDKs. Classes are defined and auto-generated using Pigeon. Mappers between Pigeon generated classes and native SmileID classes are manually written. This is a lot of LoC, due to a lot of generated code. 

## Test Instructions

Modify the sample app to run any of the networking API calls, i.e. `SmielID.api.getServices`

i.e Add a button to the sample app and verify that the response object is printed
```dart
ElevatedButton(
          child: const Text("Execute network request"),
          onPressed: () {
            SmileID.api.getServices().then((services) => {
              print("services: $services")
            }, onError: (error) => {print("error: $error")});
          },
        )
```
